### PR TITLE
Update Erga post assembly qc tools, workflow and text

### PR DESCRIPTION
**Pull request by tbrown91**
I have updated one tool here, changing the long-read mapping from hisat2 to minimap2 and updated the associated workflow.

I also changed the text a fair bit to correct some typos, add some more context to the analysis and give some more information about ERGA.

Some of the images (blob plots) had to be regenerated to reflect the new tool. Unfortunately as it stands only one dataset (the sponge) will work with the full workflow as the other datasets are too large to work with smudgeplots on galaxy. I've tested it on .eu and now that blobtoolkit is available on .org it should hopefully work there as well.

This is my first time attempting to contribute, so I'm anticipating some corrections will need to be made. For example, I tried a lot of things to make the ERGA logo smaller, but it didn't seem to change the size when I looked in gitpod.

### CODE REVIEW

**Comment by bgruening** [comment]
This look very cool, I tried to help with the merge conflicts, please always pull from upstream/main to avoid any conflicts next time.

I hope I did not screw anything up, please check my merge :)

File: topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md [hunk 1]
```diff
@@ -17,7 +17,8 @@ contributors:
 - GitFab93
 - gallardoalba
 - tbrown91
-
+tags:
```
Comment (tbrown91) [inline topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md:20]: I don't know where this tag came from and why it is plants. Should I change it to something else?

**Comment by tbrown91** [comment]
Thanks Björn and yes, sorry for not doing this on the updated main. I created my fork so long ago that it looks like a lot of things changed. From what i can see, almost everything is the same

File: topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md (hunk 1 above)
Comment (tbrown91) [reply topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md:20]: At the moment, this is what the user sees when they search for the tutorial:

image

Comment (bgruening) [reply topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md:20]: This tutorial can be used on plant genomes? Maybe add some more tags?

File: topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md [hunk 2]
```diff
@@ -575,6 +577,7 @@ gfastats is a tool for providing summary statistics and genome file manipulation
 
 To understand how chromosomes are arranged in their three-dimensional structure in the nucleus, high resolution and high throughput imaging techniques have been developed. Hi-C is a high throughput method to measure pairwise contacts between pairs of genomic loci. 
 It is based on cross-linking the DNA in the nucleus and its histones (proteins around which the DNA is wrapped). However the DNA gets cut and marked and afterwards the cut parts get ligated together. Then the DNA gets fragmented and sequenced forwards and backwards generating paired reads. The resulting information can be used to assemble the reads to the corresponding chromosome.
+
```
Comment (bgruening) [inline topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md:580]:
```suggestion
```

File: CONTRIBUTORS.yaml [hunk 3]
```diff
@@ -2435,3 +2435,13 @@ ThibaudGlinez:
...
       - elixir-europe
+
+
+tbrown91:
+    name: Tom Brown
+    email: brown@izw-berlin.de
+    orcid: 0000-0001-8293-4816
+    joined: 2024-08
+    affiliations:
+      - IZW
```
Comment (bgruening) [inline CONTRIBUTORS.yaml:2446]: Those are new to GTN and need to be defined in the https://github.com/galaxyproject/training-material/blob/main/CONTRIBUTORS.yaml

The linter complains with:

```
CONTRIBUTORS.yaml  has errors
  [/tbrown91/affiliations/0] 'IZW': invalid enum value.
  [/tbrown91/affiliations/1] 'ERGA': invalid enum value.
```

Comment (tbrown91) [reply CONTRIBUTORS.yaml]: I'm happy for now not having an affiliation - I'm not really sure I should have ERGA in the FUNDERS list

File: topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md [hunk 4]
```diff
@@ -184,29 +189,30 @@ In this tutorial, we will use BlobToolKit in order to integrate the following da
...
-> <hands-on-title>Generate BAM file with HISAT2</hands-on-title>
+> <hands-on-title>Generate BAM file with Minimap2</hands-on-title>
 >
 > 1. {% tool [Collapse Collection](toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0) %} with the following parameters:
 >    - {% icon param-collection %} *"Collection of files to collapse into single dataset"*: `CReniformis_Pacbio`
 >
 >
-> 2. {% tool [HISAT2](toolshed.g2.bx.psu.edu/repos/iuc/hisat2/hisat2/2.2.1+galaxy1) %} with the following parameters:
->    - *"Source for the reference genome"*: `Use a genome from history`
+> 2. {% tool [Map with minimap2](toolshed.g2.bx.psu.edu/view/iuc/minimap2/5cc34c3f440d) %} with the following parameters:
```
Comment (bgruening) [inline topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md:210]: This ID looks wrong. Did you mean
```suggestion
> 2. {% tool [Map with minimap2](toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.28+galaxy0) %} with the following parameters:
```

You can get this ID from the Galaxy tool from, top bar, arrow-down, tool-ID.

File: topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md [hunk 5]
```diff
@@ -3,33 +3,37 @@ layout: tutorial_hands_on
...
 tags:
   - plants
-
+  - animals
+  - genome
+  - assembly
+  - QC
 ---
 
+![](../../images/post-assembly-QC/ERGA.logo.jpg){: width="20px"}
```
Comment (bgruening) [inline topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.md:28]:
```suggestion
![ERGA log, 4 colorful letters on white background, with the European star circle](../../images/post-assembly-QC/ERGA.logo.jpg){: width="20px"}
```

File: topics/assembly/images/post-assembly-QC/erga-logo.png
Comment (bgruening) [inline topics/assembly/images/post-assembly-QC/erga-logo.png:1]: Is this logo still needed? You have 2 include.

**Comment by tbrown91** [comment]
Do you have a tutorial guide? The test dataset given from the original training material is this one:

https://zenodo.org/records/7831298

image


I assume the files are too large to be included, oder?

**Comment by bgruening** [comment]
Maybe that helps? https://github.com/galaxyproject/iwc/blob/main/workflows/README.md#generate-tests

https://planemo.readthedocs.io/en/latest/automating_workflows.html

**Comment by bgruening** [comment]
In the end we need such a `*-test.yml` file: https://github.com/galaxyproject/training-material/tree/main/topics/assembly/tutorials/unicycler-assembly/workflows

**Comment by tbrown91** [comment]
oof for you, I will give it a go.

I think the easiest for me would be to link an existing invocation on galaxy, would this be ok?

**Comment by bgruening** [comment]
Yeah, I know it's a high bar ... please link it as comment, we will try to work on this.

**Comment by tbrown91** [comment]
@bgruening how does this look? I added the urls to all the large datasets, but hopefully the rest are not too large

**Comment by SaimMomin12** [comment]
@tbrown91 in my opinion, most of the test data still should go to Zenodo - as many times it occurst that the CI/CD complains regarding exceeding file sizes

File: topics/assembly/tutorials/ERGA-post-assembly-QC/workflows/Post-assembly-workflow-tests.yml [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    log_plot (Step 36):
+      path: test-data/log_plot (Step 36).png
+    transformed_linear_plot (Step 36):
+      path: test-data/transformed_linear_plot (Step 36).png
+    transformed_log_plot (Step 36):
+      path: test-data/transformed_log_plot (Step 36).png
+    linear_plot (Step 36):
+      path: test-data/linear_plot (Step 36).png
+    stats (Step 13):
+      path: test-data/stats (Step 13).tabular
```
Comment (SaimMomin12) [inline topics/assembly/tutorials/ERGA-post-assembly-QC/workflows/Post-assembly-workflow-tests.yml:52]: Maybe adding some asserts would be beneficial here?

File: topics/assembly/tutorials/ERGA-post-assembly-QC/workflows/Post-assembly-workflow-tests.yml [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      collection_type: list
+      elements:
+      - class: File
+        identifier: '0'
+        path: https://zenodo.org/record/7786773/files/hiC_1.fastq.gz
+    NCBI taxonomic ID: '121349'
+    Ploidy for model to use: '2'
+  outputs:
+    log_plot (Step 36):
+      path: test-data/log_plot (Step 36).png
```
Comment (SaimMomin12) [inline topics/assembly/tutorials/ERGA-post-assembly-QC/workflows/Post-assembly-workflow-tests.yml:44]: PNGS and JPGs can be moved to Zenodo. Else, everytime you need to compress them in order to reduce file size

File: topics/assembly/tutorials/ERGA-post-assembly-QC/workflows/Post-assembly-workflow-tests.yml [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    smudgeplot (Step 21):
+      path: test-data/smudgeplot (Step 21).png
+    smudgeplot_log (Step 21):
+      path: test-data/smudgeplot_log (Step 21).png
+    outfile (Step 23):
+      path: test-data/outfile (Step 23).jpg
+    'Busco on input dataset(s): full table':
+      path: 'test-data/Busco on input dataset(s): full table.tabular'
+    'Busco on input dataset(s): short summary':
+      path: 'test-data/Busco on input dataset(s): short summary.txt'
```
Comment (SaimMomin12) [inline topics/assembly/tutorials/ERGA-post-assembly-QC/workflows/Post-assembly-workflow-tests.yml:62]: Perhaps adding asserts to files other than PNGs and JPGs

**Comment by SaimMomin12** [comment]
Maybe this helps to give you an idea about asserts in the workflow test 

https://github.com/galaxyproject/training-material/blob/4ab5654df7903a998f5c6ccfaef7c1db8d3d162a/topics/microbiome/tutorials/dada-16S/workflows/workflow-tests.yml#L213

**Comment by tbrown91** [comment]
Hi @SaimMomin12 Thanks for the pointers. 

Just to make sure, I can put all of the images on zenodo and put a url to the images in the .yml?

I think the txt files output from this workflow are quite simple, so I should be able to write quite strict asserts. I will have a go at this today

**Comment by SaimMomin12** [comment]
@tbrown91 The other thing which you can do is to "compress" the images and place them in the folder below in order to have them locally in the repo itself

https://github.com/galaxyproject/training-material/tree/main/topics/assembly/images/post-assembly-QC/

**Comment by tbrown91** [comment]
Does this look correct?

I annoyingly only saw after publishing that the names have spaces in:

image

https://zenodo.org/records/13627253

**Comment by SaimMomin12** [comment]
Looks good to me!

**Comment by tbrown91** [comment]
Perfect, see what you think. Some of the files have differing numbers of columns, but hopefully the rest will work

**Comment by tbrown91** [comment]
well this is fun...

**Comment by tbrown91** [comment]
I see these errors, which I hope are easy to fix:

```"There are colons in this filename, that is forbidden.```

and this, which I hope is a result of the above error:

```This workflow is missing a test, which is now mandatory.```

is that correct?

**Comment by bgruening** [comment]
Yes, think so. The spaces in the filename are not good and can cause problems. Let us know when you are running out of time and we should take over. Thanks a lot Tom!

**Comment by tbrown91** [comment]
So do you think I need to remove all of the spaces from the file names, or will these be ok now?

**Comment by bgruening** [comment]
sorry, since its your fist contribution here, I always needs to confirm the CI - a security feature from Github. And I'm slow, especially on vacation.

**Comment by tbrown91** [comment]
Bjoern, you have to tell me if you are on holiday, and I will stop bugging you

File: topics/assembly/tutorials/ERGA-post-assembly-QC/workflows/test-data/Busco on input dataset(s) full table.tabular
Comment (bgruening) [inline topics/assembly/tutorials/ERGA-post-assembly-QC/workflows/test-data/Busco on input dataset(s) full table.tabular:1]: Is this one still used? Its has spaces in the file name.

**Review by bgruening — COMMENTED** [review]
Its green! Do you mind changing the filenames in this repo to also remove spaces and brackets? I think then we are ready.

**Comment by tbrown91** [comment]
thanks for your help

**Comment by tbrown91** [comment]
@SaimMomin12 should I also remove the spaces from the names of the files on zenodo, or are these ok?
e.g. https://zenodo.org/records/13639718/files/transformed_log_plot%20(Step%2036).png

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@tbrown91 spaces are always bad ;)

Please remove them when you update the tutorial next time :)


Thanks a lot! This is a very nice tutorial!

**Pull request merged**